### PR TITLE
sgtk-menu: init at 1.4.1

### LINKED ISSUE #78370: sgtk-menu
**Project description**
Simple menu, that behaves decently on sway, but also on i3 window manager.

**Metadata**

* homepage URL: https://github.com/nwg-piotr/sgtk-menu
* source URL: https://github.com/nwg-piotr/sgtk-menu
* license: gpl3 https://github.com/nwg-piotr/sgtk-menu/blob/master/LICENSE
* platforms: linux

### PR DESCRIPTION
**Pull request by berbiche**
###### Motivation for this change

Packages sgtk-menu. Fixes #78370.

The wiki mentions the package `python-pynput` is needed to spawn the widget at the pointer location but `python-pynput` is not currently packaged.

cc @eoli3n if you want to test this out.

When testing the program, right clicks made `sgtk-grid` exit successfully without launching any application.
I don't know whether this is intended behavior since I've never used the program before.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by berbiche** [comment]
This package is also relevant for #57602

**Comment by pbogdan** [comment]
Just a note that there is also an open PR for packaging the C++ version - https://github.com/NixOS/nixpkgs/pull/93189

File: pkgs/applications/misc/sgtk-menu/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  buildInputs = [ gtk3 gobject-introspection librsvg gdk-pixbuf glib ];
+
+  propagatedBuildInputs = [ setuptools pygobject3 pycairo ];
+
+  # Required for librsvg & gdk-pixbuf to work
+  strictDeps = false;
+
+  meta = with stdenv.lib; {
+    description = "A themeable, searchable, gtk3-based system launcher.";
```
Comment (symphorien) [inline pkgs/applications/misc/sgtk-menu/default.nix:30]:
```suggestion
    description = "A themeable, searchable, gtk3-based system launcher";
```
As per https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md

**Comment by symphorien** [comment]
When clicking on the `lock` button, it seems that it depends on swaylock. I wonder if the path should be hardcoded.

**Comment by berbiche** [comment]
> When clicking on the `lock` button, it seems that it depends on swaylock. I wonder if the path should be hardcoded.

Good point. Swaylock is only used when sgtk-menu detects that it's running on Sway.

I don't know whether `substituteAll` is more appropriate than setting the path in the wrapper.

**Comment by symphorien** [comment]
> Good point. Swaylock is only used when sgtk-menu detects that it's running on Sway.

well it does not detect right because I tested on i3.

**Comment by berbiche** [comment]
> > Good point. Swaylock is only used when sgtk-menu detects that it's running on Sway.
> 
> well it does not detect right because I tested on i3.

This is the code responsible for detecting your WM
https://github.com/nwg-piotr/sgtk-menu/blob/b253f8cc6b025635125b4b9e4a6023c20b4b8e35/sgtk_menu/tools.py#L20-L77

> Just a note that there is also an open PR for packaging the C++ version - https://github.com/NixOS/nixpkgs/pull/93189

Judging by the code activity on sgtk-menu, it will eventually be archived.

I will close this PR in favor of #93189.

**Pull request closed without merging**